GSoC-2025/movement: front-end support for filtering module.md

```diff
@@ -0,0 +1,112 @@
+  4. [Added a Widget for drawing region of interest in napari (feature discussed in #378).](https://github.com/neuroinformatics-unit/movement/pull/489) 
+  5. [Fix Runtime Warning raised by ROI plot test](https://github.com/neuroinformatics-unit/movement/pull/534)
+- **Proposal discussion link**:   
+
+---
+
+## Project Proposal  
+**Synopsis**  
+This project aims to develop an intuitive napari plugin for Movement’s filtering module, enabling non-programmers to clean and analyze motion-tracking data (e.g., from pose estimation tools like DeepLabCut) through a graphical interface instead of Python scripting.  
+
+**Why It Matters**  
+- **Accessibility Gap**: Most biologists/neuroscientists lack coding skills but rely on pose-tracking data.  
+- **Time Efficiency**: Manual data cleaning consumes research time; real-time filtering previews streamline workflows.  
+- **Reproducibility**: GUI-driven workflows ensure standardized filtering across labs.  
+- **Open Science**: Democratizes advanced tools for underrepresented institutions with limited computational resources.  
+
+---
+
+## Implementation Timeline  
+### 1. **Core Deliverables**  
+- **Napari Widget Suite**:  
+  - Dropdown menus for confidence filters (threshold-based outlier removal).  
+  - Sliders/inputs for Median filter (window size) and SavGol filter (window size, polynomial order).  
+  - Real-time overlay of raw vs. filtered data.  
+- **Testing Framework**:  
+  - Unit tests for filter logic and UI components.  
+  - Performance benchmarks.  
+- **Documentation**:  
+  - Jupyter notebook tutorials with sample datasets.  
+  - Contributor guidelines for extending the widget.  
+- **Export Workflows**: Save/load filter configurations as netCDF.  
+- **Video Tutorial**: 5-minute demo for all the filters, covering how and when to use each. 
+
+### 2. **Extended Deliverables (Aspirational)**  
+- **Filter Chaining**: Sequential application of filters (e.g., confidence → median → SavGol).  
+- **Community Presets**: Curated settings for common use cases (e.g., fruit fly locomotion).  
+- **Hardware Acceleration**: GPU-backed filtering.  
+
+---
+
+## GSoC 2025 Timeline  
+### **Community Bonding Period** *(May 8 – June 1)*  
+| Week | Tasks | Hours | Output |  
+|------|-------|-------|--------|  
+| CB1  | Study napari plugin architecture; draft wireframes with mentors | 7 | Environment setup guide |  
+| CB2  | Profile filter performance; create UML diagrams; setup Sphinx docs | 8 | Technical design doc |  
+| CB3  | Finalize widget CSS theme; draft Jupyter notebook | 5 | PR for docs infrastructure |  
+
+### **Coding Period** *(June 2 – August 24)*  
+
+| Week | Focus Area | Critical Deliverables | Hours |  
+|------|------------|------------------------|-------|  
+| 1    | Widget Framework | Base widget class; confidence threshold UI | 15 |  
+| 2    | Median Filter | Rolling window logic; unit tests | 14 |  
```

Comment:
note that this has now been updated to `rolling_filter`, with "median", "mean", "min", and "max" being various choices.